docs/documentation/features/frameworkUtilities.md

```diff
@@ -6,9 +6,116 @@ layout: documentation
+given context is not configured.
+
+If nothing is configured (first-start), the service will generate a self-signed certificate and public/private key-pair in `ESH_DIR/etc/default.keystore` and store a new configuration file in `ESH_DIR/etc/networkServerTlsProvider.cfg`, including the keystore access password and private key password.
+
+You can provide your own java keystore files in `ESH_DIR/etc/yourKeystoreFile.keystore`. The name of the file is automatically the context name, e.g. you would request a `SSLContext` like this: `createSSLContext("yourKeystoreFile")`.
+Don't forget to add the configuration for your context to the configuration file as explained in the next section.
+
+The service can use ACME to automatically sign a public key by the Let's Encrypt initiative. 
+
+### Configuration
+
+The service accepts a configuration file `networkServerTlsProvider.cfg` with the following parameters:
+
+* **domains**: for instance "www.abc.org|default,www.def.org|second". *¹ *²
+* **psks**: for instance "JVBER|default,i0xLjUK|second". *¹ *²
+* **keystorePasswords**: for instance "JVBER|default,i0xLjUK|second". *¹ *²
+* **privateKeyPasswords**: for instance "JVBER|default,i0xLjUK|second". *¹ *²
+
+   *¹: Multiple contexts possible, separated by comma. The context name is appended with the pipe character ("|").
+   *²: Base64 encoded password. Your password can contain any byte values and must be Base64 encoded to make it human read- and writable in the editor. You can create a base64 encoded string of your password on the command line of Linux/MacOS with `echo "abcdef" |base64` with _abcdef_ being your password.
+  
+Please be aware that if no public/private key and certificate can be found for the default context, the
+service will create a configuration file or overwrite an existing one after creating the public/private key and certificate file.
+
+For generating a self-signed certificate on the first start, the following parameters will be used:
+
+* **organizationalUnit**: "Eclipse Smarthome"
+* **organization**: "Eclipse Foundation, Inc."
+* **city**: "Ottawa"
+* **state**: "Ontario"
+* **country**: for instance "US", the java environment will be asked for the current country code.
+* **validityInDays**: 365 * 3
+
+The following algorithm specific parameters can be set:
+
+* **keysizeTLS**: 2048
+* **keysizeDTLS**: 256
+* **algorithmTLS**: RSA
+* **algorithmDTLS**: EC
+* **algorithmSigningTLS**: SHA1WithRSA
+* **algorithmSigningDTLS**: SHA256withECDSA
+
+And those management configurations are supported:
+
+* **autoRefreshInvalidCertificate**: Default is false. Automatically create a new self-signed certificate after the current one expired. Only happens for the requested context, other certificates, even if also expired, are kept. Please be aware that your clients may need to be deployed with the new public key.
+* **useACME**: Default is false. ACME is used to sign generated public/private keys.
```

Comment:
The plan was, to prepare the service and all necessary configuration to support ACME and implement it in a follow up PR. Would be worse, if we need to change the API later on, because ACME needs something specific.